front/lib/api/assistant/messages.ts

```diff
@@ -213,20 +277,38 @@ async function batchRenderAgentMessages<V extends RenderMessageVariant>(
         };
       }
 
+      const stepContents = await AgentStepContentResource.fetchByAgentMessages(
```

Comment:
I don't think we want to fetch this in a loop. Can we fetch them for all messages at once above ?